dali/pipeline/operators/color/hsv.h

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_COLOR_HSV_H_
+#define DALI_PIPELINE_OPERATORS_COLOR_HSV_H_
+
+#include <vector>
+#include <memory>
+#include <string>
+#include "dali/core/static_switch.h"
+#include "dali/core/geom/mat.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operators/common.h"
+#include "dali/pipeline/operators/operator.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/kernels/tensor_shape_print.h"
+#include "dali/kernels/imgproc/pointwise/linear_transformation_cpu.h"
+
+
+namespace dali {
+namespace hsv {
+
+/**
+ * Names of arguments
+ */
+extern const std::string kHue;
+extern const std::string kSaturation;
+extern const std::string kValue;
+extern const std::string kOutputType;
+
+/**
+ * Color space conversion
+ */
+const mat3 Rgb2Yiq = {{
```

Comment:
A global variable in a header file - place it in anonymous namespace or convert to a constexpr function.

Reply:
Not good enough for the linker.

Reply:
Sorry, my bad - apparently, C++11 makes this "internal linkage". I remember having problems with it, though...

Reply:
Resolved then